fmn/delivery/service.py

```diff
@@ -0,0 +1,242 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of the FMN project.
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+"""
+The FMN message delivery service.
+
+This service is a `Twisted`_ application which consumes messages from an AMQP queue. Those
+messages are then delivered using a back-end such as email or IRC. Messages are sent to
+the queue by :func:`fmn.tasks.find_recipients` and :func:`fmn.tasks.batch_messages`.
+
+This service is intended to be run using ``twistd`` and a Twisted application (.tac) file.
+
+.. _Twisted: https://twistedmatrix.com/
+"""
+from __future__ import absolute_import
+
+import logging
+
+from kombu import Connection, Queue
+from kombu.mixins import ConsumerMixin
+from kombu.pools import connections
+from twisted.application import service
+from twisted.internet import threads, reactor, task, defer
+import six
+
+from fmn import config
+from fmn.lib import models
+from .backends import sse, irc, debug, mail
+
+_log = logging.getLogger(__name__)
+
+
+class Consumer(ConsumerMixin):
+    """
+    A Kombu Consumer that ferries messages from AMQP into the Twisted reactor thread
+    via a blocking API.
+
+    This is intended to be run in its own thread using Twisted's :func:`reactor.callInThread`
+    API. It will then use the :func:`threads.blockingCallFromThread` API to deliver messages
+    into the Twisted reactor thread.
+
+    Args:
+        delivery_service (DeliveryService): Instance of the Twisted delivery service to use when
+            dispatching messages to be sent inside the Twisted reactor thread.
+        queues (list): A list of queues to consume from.
+    """
+
+    def __init__(self, delivery_service, queues):
```

Comment:
Are the Args above meant to be documented here?

Reply:
This style allows you to document the ``__init__`` function in the class definition if you use Args (see http://www.sphinx-doc.org/en/1.6.3/ext/example_google.html)